homeassistant/components/nest/climate_sdm.py

```diff
@@ -319,16 +314,12 @@ async def async_set_preset_mode(self, preset_mode):
         """Set new target preset mode."""
         if preset_mode not in self.preset_modes:
             return
-        if ThermostatEcoTrait.NAME not in self._device.traits:
-            return
         trait = self._device.traits[ThermostatEcoTrait.NAME]
         await trait.set_mode(PRESET_INV_MODE_MAP[preset_mode])
 
     async def async_set_fan_mode(self, fan_mode):
         """Set new target fan mode."""
         if fan_mode not in self.fan_modes:
             return
-        if FanTrait.NAME not in self._device.traits:
```

Comment:
Same as the above comment, shouldn't this throw a valuerror when an invalid fan mode is set?

Reply:
Updated.